src/ol_infrastructure/components/services/k8s.py

```diff
@@ -0,0 +1,311 @@
+                                        container_port=DEFAULT_NGINX_PORT
+                                    )
+                                ],
+                                image_pull_policy="IfNotPresent",
+                                resources=kubernetes.core.v1.ResourceRequirementsArgs(
+                                    requests={"cpu": "50m", "memory": "64Mi"},
+                                    limits={"cpu": "100m", "memory": "128Mi"},
+                                ),
+                                volume_mounts=[
+                                    kubernetes.core.v1.VolumeMountArgs(
+                                        name="staticfiles",
+                                        mount_path="/src/staticfiles",
+                                    ),
+                                    kubernetes.core.v1.VolumeMountArgs(
+                                        name="nginx-config",
+                                        mount_path="/etc/nginx/conf.d/web.conf",
+                                        sub_path="web.conf",
+                                        read_only=True,
+                                    ),
+                                ],
+                            ),
+                            # Actual application run with uwsgi
+                            kubernetes.core.v1.ContainerArgs(
+                                name="application-app",
+                                image=f"mitodl/unified-application-app-main:{self.ol_app_k8s_config.application_docker_tag}",
+                                ports=[
+                                    kubernetes.core.v1.ContainerPortArgs(
+                                        container_port=DEFAULT_UWSGI_PORT
+                                    )
+                                ],
+                                image_pull_policy="IfNotPresent",
+                                resources=kubernetes.core.v1.ResourceRequirementsArgs(
+                                    requests={"cpu": "250m", "memory": "300Mi"},
+                                    limits={"cpu": "500m", "memory": "600Mi"},
+                                ),
+                                env=application_deployment_env_vars,
+                                env_from=application_deployment_envfrom,
+                                volume_mounts=[
+                                    kubernetes.core.v1.VolumeMountArgs(
+                                        name="staticfiles",
+                                        mount_path="/src/staticfiles",
+                                    ),
+                                ],
+                            ),
+                        ],
+                    ),
+                ),
+            ),
+            opts=resource_options,
+        )
+
+        # A kubernetes service resource to act as load balancer for the app instances
+        _application_service = kubernetes.core.v1.Service(
```

Comment:
@Ardiea Is this actually part of the APISix stuff I should be leaving in unified_ecommerce and not including in my component for now?

I ask because this blob necessitates the load balancer service name and port name which I needed to add to my configuration class and they made me wonder if they're out of place.